### PR TITLE
Page.uploadFile

### PR DESCRIPTION
**Pull request by JoelEinbinder**
#54

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -236,6 +237,11 @@ Pages could be closed by `page.close()` method.
 
 #### page.type()
 
+#### page.uploadFile(selector, ...files)
+- `selector` <[string]> A query selector to a file input
+- `...files` <[string]> Sets the value of the file input these files
```
Comment (paulirish) [inline docs/api.md]: I was definitely expecting this to be taking a file descriptor or buffer.  how about

> Sets the value of the file input these file paths

and i would change the name of the argument to `filePaths`

Comment (JoelEinbinder) [reply docs/api.md]: Done.

File: lib/Page.js [hunk 2]
```diff
@@ -569,6 +569,13 @@ class Page extends EventEmitter {
   waitFor(selector) {
     return this.mainFrame().waitFor(selector);
   }
+
+  async uploadFile(selector, ...filePaths) {
+    await this._client.send('DOM.setFileInputFiles', {
+      nodeId: await this._querySelector(selector),
```
Comment (aslushnikov) [inline lib/Page.js]: what will happen if this._querySelector fails?

File: lib/Page.js [hunk 3]
```diff
@@ -569,6 +569,13 @@ class Page extends EventEmitter {
   waitFor(selector) {
     return this.mainFrame().waitFor(selector);
   }
+
+  async uploadFile(selector, ...filePaths) {
```
Comment (aslushnikov) [inline lib/Page.js]: let's add jsdoc

File: phantom_shim/WebPage.js [hunk 4]
```diff
@@ -333,6 +333,17 @@ class WebPage {
   }
 
   /**
+   * @param {string} selector
+   * @param {string|!Array<string>} files
```
Comment (aslushnikov) [inline phantom_shim/WebPage.js]: i'm pretty sure type annotation should be {(string|!Array<string>)}

File: lib/Page.js (hunk 3 above)
Comment (aslushnikov) [reply lib/Page.js]: I don't mind having `...filePaths`, but curious why you chose this over the `array<string>`, which seems to be more natural?

File: phantom_shim/WebPage.js (hunk 4 above)
Comment (JoelEinbinder) [reply phantom_shim/WebPage.js]: But it's an array of strings?

File: lib/Page.js (hunk 3 above)
Comment (JoelEinbinder) [reply lib/Page.js]: Multiple file upload is an edge case, so i wanted to have ```uploadFile('input', 'myfile.txt')``` work.

File: phantom_shim/WebPage.js (hunk 4 above)
Comment (aslushnikov) [reply phantom_shim/WebPage.js]: Right, I meant `{(string|!Array<string>)}`

File: lib/Page.js (hunk 2 above)
Comment (JoelEinbinder) [reply lib/Page.js]: Protocol will throw an error. Same will happen with`.click`.

File: phantom_shim/WebPage.js (hunk 4 above)
Comment (JoelEinbinder) [reply phantom_shim/WebPage.js]: I don't understand, but sure.

File: lib/Page.js (hunk 3 above)
Comment (JoelEinbinder) [reply lib/Page.js]: Done.

File: lib/Page.js (hunk 2 above)
Comment (aslushnikov) [reply lib/Page.js]: Let's handle this separately and throw a more clear error. Currently we get
```
(node:30617) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Protocol error (DOM.getBoxModel): Could not find node with given id undefined
```
which is quite confusing.

**Pull request merged**